### DISCUSSION TITLE
Show loading while doing .configure work

### DISCUSSION THREAD

**Discussion by Bjorninn1** [airbnb/lottie-ios#2195] [Q&A] [answered]
I have been trying out the new LottieView. Is there a way to show loading somehow while code in .configure is running? 
I tried the following, but it causes an endless loop. This is as expected since state is being updated inside the view, but what would be the correct way to implement this or something similar? The problem is that animationView seems to be only accessible inside .configure, so not sure how to get around this. All help and ideas appreciated.

```
LottieView(animation: LottieAnimation.named(imageName))
.configure { animationView in
    DispatchQueue.main.async {
        self.isLoading = true
    }

    //some work

    DispatchQueue.main.async {
        self.isLoading = false
    }

}
```

**Comment by calda** [comment, marked as answer]
`.configure` is called on every SwiftUI state update so shouldn't be used for anything stateful like this. What are you trying to do exactly?

One way to show a loading indicator is using this `init` syntax:

```swift
LottieView {
  LottieAnimation.named(imageName)
} placeholder: {
  // Displayed until the animation is loaded
  MyLoadingIndicator()
}
```

Alternatively you could do something like:

```swift
LottieView(animation: LottieAnimation.named(imageName))
  .animationDidLoad {
    // The animation has now finished loading. Called exactly once.
  }
```
